frontend/actions/src/main/java/com/virtuslab/gitmachete/frontend/actions/base/BaseSquashAction.java

```diff
@@ -154,7 +154,7 @@ public void run(ProgressIndicator indicator) {
               LOG.info("Checking out '${branchName}' branch and squashing it");
 
               if (!isSquashingCurrentBranch) {
-                CheckoutSelectedAction.doCheckout(project, indicator, branchName, gitRepository);
+                CheckoutBackgroundable.doCheckout(project, indicator, branchName, gitRepository);
```

Comment:
Hmmm this method, despite being executed on a `...Backgroundable` class, isn't really executed in a background task, right? 🤔 this is somewhat confusing

Reply:
Maybe, just `CheckoutBackgrdundable` itself should be constructed and then `.queue` should be called on that. 

It seems that the visibility of the `doCkeckout` method which is called in the the `run` method of the `CheckoutBackgrdundable` should be changed to `private`

Reply:
fixed by removing this class :trollface:

Reply:
> fixed by removing this class :trollface:

It is needed by Traverse.